src/common/serializer/binary_serializer.cpp

```diff
@@ -46,21 +81,26 @@ void BinarySerializer::OnMapEnd(idx_t count) {
 void BinarySerializer::OnObjectBegin() {
 	stack.push_back(State({0, 0, data.size()}));
 	// Store the field id
-	Write<field_id_t>(current_field_id);
+	Write<uint32_t>(current_field_id);
```

Comment:
`field_id_t`

Reply:
I am not sure I agree, Write<> should be called only by trivial C++ types (and not all of them), so maybe it's more clear going with uint32_t (given this will become a constraint for storage).